entities-viewings-collector/src/test/scala/io/renku/entities/viewings/search/DatasetQuerySpec.scala

```diff
@@ -0,0 +1,137 @@
+    upload(projectsDataset, person)
+    provisionTestProjects(project1, project2).unsafeRunSync()
+
+    storeProjectViewed(userId, Instant.now().minus(1, ChronoUnit.DAYS), project1.path)
+    storeDatasetViewed(userId, Instant.now().minus(2, ChronoUnit.DAYS), dataset1.identification.identifier)
+    storeProjectViewed(userId, Instant.now().minus(3, ChronoUnit.DAYS), project2.path)
+    storeDatasetViewed(userId, Instant.now().minus(4, ChronoUnit.DAYS), dataset2.identification.identifier)
+
+    val query = DatasetQuery.makeQuery(Criteria(Set.empty, AuthUser(userId, token), 5))
+
+    val decoded = tsClient.queryExpecting[List[SearchEntity.Dataset]](query)(datasetDecoder).unsafeRunSync()
+
+    decoded.size shouldBe 2
+  }
+
+  it should "find and decode datasets" in {
+    val project = renkuProjectEntities(visibilityPublic)
+      .withActivities(activityEntities(stepPlanEntities()))
+      .withDatasets(datasetEntities(provenanceNonModified))
+      .generateOne
+
+    val dataset = project.to[entities.RenkuProject.WithoutParent].datasets.head
+
+    val person = personGen.generateOne
+    upload(projectsDataset, person)
+
+    val userId = person.maybeGitLabId.get
+
+    provisionTestProjects(project).unsafeRunSync()
+    storeDatasetViewed(userId, Instant.now(), dataset.identification.identifier)
+
+    val query = DatasetQuery.makeQuery(Criteria(Set.empty, AuthUser(userId, token), 5))
+
+    val decoded = tsClient.queryExpecting[List[SearchEntity.Dataset]](query)(datasetDecoder).unsafeRunSync()
+    decoded.head shouldMatchTo
+      SearchEntity.Dataset(
+        sameAs = dataset.provenance.topmostSameAs,
+        matchingScore = MatchingScore(1f),
+        name = dataset.identification.name,
+        visibility = project.visibility,
+        date = dataset.provenance.date,
+        keywords = dataset.additionalInfo.keywords.sorted,
+        maybeDescription = dataset.additionalInfo.maybeDescription,
+        images = dataset.additionalInfo.images.map(_.uri),
+        creators = dataset.provenance.creators.toList.map(_.name),
+        exemplarProjectPath = project.path
+      )
+  }
+
+  it should "return datasets for the given user only" in {
+    val project1 = renkuProjectEntities(visibilityPublic)
+      .withActivities(activityEntities(stepPlanEntities()))
+      .withDatasets(datasetEntities(provenanceNonModified))
+      .generateOne
+    val project2 = renkuProjectEntities(visibilityPublic)
+      .withActivities(activityEntities(stepPlanEntities()))
+      .withDatasets(datasetEntities(provenanceNonModified))
+      .generateOne
+
+    val person1 = personGen.generateOne
+    val person2 = personGen.generateOne
+    upload(projectsDataset, person1, person2)
+    provisionTestProjects(project1, project2).unsafeRunSync()
+
+    val dataset1 = project1.to[entities.RenkuProject.WithoutParent].datasets.head
+    val dataset2 = project2.to[entities.RenkuProject.WithoutParent].datasets.head
+    val userId1  = person1.maybeGitLabId.get
```

Comment:
It's not a big deal but if we'd mix in `OptionValues` and use `.value` instead of `.get` we can eventually get slightly more informative failures when they happen.